server/src/org/apache/cloudstack/network/lb/CertServiceImpl.java

```diff
@@ -339,190 +348,182 @@ public SslCertResponse createCertResponse(SslCertVO cert, List<LoadBalancerCertM
-        } catch (NoSuchPaddingException e) {
+        } catch (final NoSuchPaddingException e) {
             throw new IllegalArgumentException("Bad public-private key", e);
-        } catch (InvalidKeyException e) {
+        } catch (final InvalidKeyException e) {
             throw new IllegalArgumentException("Invalid public-private key", e);
-        } catch (NoSuchAlgorithmException e) {
+        } catch (final NoSuchAlgorithmException e) {
             throw new IllegalArgumentException("Invalid algorithm for public-private key", e);
         }
     }
 
-    private void validateChain(List<Certificate> chain, Certificate cert) {
+    private void validateChain(final List<Certificate> chain, final Certificate cert) {
 
-        List<Certificate> certs = new ArrayList<Certificate>();
-        Set<TrustAnchor> anchors = new HashSet<TrustAnchor>();
+        final List<Certificate> certs = new ArrayList<Certificate>();
+        final Set<TrustAnchor> anchors = new HashSet<TrustAnchor>();
 
         certs.add(cert); // adding for self signed certs
         certs.addAll(chain);
 
-        for (Certificate c : certs) {
-            if (!(c instanceof X509Certificate))
+        for (final Certificate c : certs) {
+            if (!(c instanceof X509Certificate)) {
                 throw new IllegalArgumentException("Invalid chain format. Expected X509 certificate");
+            }
 
-            X509Certificate xCert = (X509Certificate)c;
+            final X509Certificate xCert = (X509Certificate)c;
 
-            Principal subject = xCert.getSubjectDN();
-            Principal issuer = xCert.getIssuerDN();
+            xCert.getSubjectDN();
+            xCert.getIssuerDN();
 
-           anchors.add(new TrustAnchor(xCert, null));
+            anchors.add(new TrustAnchor(xCert, null));
         }
 
-        X509CertSelector target = new X509CertSelector();
+        final X509CertSelector target = new X509CertSelector();
         target.setCertificate((X509Certificate)cert);
 
         PKIXBuilderParameters params = null;
         try {
             params = new PKIXBuilderParameters(anchors, target);
             params.setRevocationEnabled(false);
             params.addCertStore(CertStore.getInstance("Collection", new CollectionCertStoreParameters(certs)));
-            CertPathBuilder builder = CertPathBuilder.getInstance("PKIX", "BC");
+            final CertPathBuilder builder = CertPathBuilder.getInstance("PKIX", "BC");
             builder.build(params);
 
-        } catch (InvalidAlgorithmParameterException e) {
+        } catch (final InvalidAlgorithmParameterException e) {
             throw new IllegalArgumentException("Invalid certificate chain", e);
-        } catch (CertPathBuilderException e) {
+        } catch (final CertPathBuilderException e) {
             throw new IllegalArgumentException("Invalid certificate chain", e);
-        } catch (NoSuchAlgorithmException e) {
+        } catch (final NoSuchAlgorithmException e) {
```

Comment:
Please consider consolidating lines 434-439 into a multi-catch.  Also, `IllegalArgumentException` seems inappropriate as it indicates that the preconditions of the method were violated.  `IllegalStateException` seems more appropriate.